addons/mail_group/models/mail_group.py

```diff
@@ -0,0 +1,435 @@
+        if any(group.moderation_notify and not group.moderation_notify_msg for group in self):
+            raise ValidationError(_('The notification message is missing.'))
+
+    @api.constrains('moderation_guidelines', 'moderation_guidelines_msg')
+    def _check_moderation_guidelines(self):
+        if any(group.moderation_guidelines and not group.moderation_guidelines_msg for group in self):
+            raise ValidationError(_('The guidelines description is missing.'))
+
+    @api.constrains('moderator_ids', 'moderation')
+    def _check_moderator_existence(self):
+        if any(not group.moderator_ids for group in self if group.moderation):
+            raise ValidationError(_('Moderated group must have moderators.'))
+
+    @api.constrains('privacy', 'group_public_id')
+    def _check_privacy(self):
+        if any(group.privacy == 'groups' and not group.group_public_id for group in self):
+            raise ValidationError(_('The "Authorized Group" is missing.'))
+
+    @api.model
+    def message_new(self, msg_dict, custom_values=None):
+        """Add the method to make the mail gateway flow work with this model."""
+        return
+
+    @api.model
+    def message_update(self, msg_dict, update_vals=None):
+        """Add the method to make the mail gateway flow work with this model."""
+        return
+
+    def message_post(self, **kwargs):
+        self.ensure_one()
+        # First create the <mail.message>
+        Mailthread = self.env['mail.thread']
+        values = dict((key, val) for key, val in kwargs.items() if key in self.env['mail.message']._fields)
+        author_id, email_from = Mailthread._message_compute_author(kwargs.get('author_id'), kwargs.get('email_from'), raise_exception=True)
+        values.update({
+            'author_id': author_id,
+            'email_from': email_from,
+            'model': self._name,
+            'res_id': self.id,
+            'partner_ids': [],
+        })
+        attachments = kwargs.get('attachments') or []
+        attachment_ids = kwargs.get('attachment_ids') or []
+        attachement_values = Mailthread._message_post_process_attachments(attachments, attachment_ids, values)
+        values.update(attachement_values)
+        mail_message = Mailthread._message_create(values)
+
+        # Create the group message associated
+        group_message = self.env['mail.group.message'].create({
+            'mail_group_id': self.id,
+            'mail_message_id': mail_message.id,
+        })
+
+        # Check the moderation rule to determine if we should accept or reject the email
+        email_normalized = email_normalize(kwargs.get('email_from'))
+        moderation = self.env['mail.group.moderation'].search([
+            ('mail_group_id', '=', self.id),
+            ('email', '=', email_normalized),
+        ], limit=1)
+
+        if not self.moderation or (moderation and moderation.status == 'allow'):
+            group_message.action_accept()
```

Comment:
But it means that you will have message in your database which will be marked as "pending moderation" but are actually sent... So if you activate the moderation after a couple of month of use, you will have a lot of message "pending moderation" that are already sent :/

Or we should call `send_message` then write `accepted` in the status, but that's exactly what `action_moderate_accept` does (so I just prefer to call `action_moderate_accept`)

Reply:
If group does not use moderation, messages should not be pending moderation. They are automatically accepted, and I think we should simply skip the whole moderation fields and steps to avoid writing unnecessary values. Moderation is quite advanced, most mailing lists will not be moderated I think.

Reply:
Ok, so I set the moderation_status to accepted in the message_post if the group is not moderated